### PR TITLE
Get the configuration of quality gates from the UI

### PR DESCRIPTION
**Pull request by jchen-parasoft**

### CODE REVIEW

**Comment by dtang-parasoft** [comment]
I'm reviewing this

File: CodeCoverageQualityGate/CodeCoverageQualityService.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    constructor() {
+        this.typeString = tl.getInput('type') || '';
+        this.thresholdString = tl.getInput('threshold') || '';
+        this.buildStatusString = tl.getInput('buildStatus') || '';
+
+        this.threshold = parseFloat(this.thresholdString || '0.0');
+
+        if (isNaN(this.threshold)) {
+            tl.warning('Illegal threshold value \'' + this.thresholdString + '\', using default value 0.0');
```
Comment (rwang-parasoft) [inline CodeCoverageQualityGate/CodeCoverageQualityService.ts:45]: Change Illegal to Invalid.

Comment (rwang-parasoft) [reply CodeCoverageQualityGate/CodeCoverageQualityService.ts:45]: Use the format string like: `Invalid threshold value '${this.thresholdString}', using default value 0.0`. The same change need to be done for other places.

File: CodeCoverageQualityGate/CodeCoverageQualityService.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            this.threshold = 0.0;
+        } else if (this.threshold > 100) {
+            tl.warning('The threshold value \'' + this.thresholdString + '\' is more than 100, the value will be cast to 100.0');
```
Comment (rwang-parasoft) [inline CodeCoverageQualityGate/CodeCoverageQualityService.ts:48]: Change 'the value will be cast to 100.0' to 'the value is cast to 100.0'

File: CodeCoverageQualityGate/CodeCoverageQualityService.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            this.threshold = 100.0;
+        } else if (this.threshold < 0) {
+            tl.warning('The threshold value \'' + this.thresholdString + '\' is less than 0, the value will be cast to 0.0');
```
Comment (rwang-parasoft) [inline CodeCoverageQualityGate/CodeCoverageQualityService.ts:51]: Change 'the value will be cast to 0.0' to 'the value is cast to 0.0'

**Comment by dtang-parasoft** [comment]
Ready to merge

File: CodeCoverageQualityGate/package.json [hunk 4]
```diff
@@ -21,5 +21,16 @@
   "bugs": {
     "url": "https://github.com/parasoft/parasoft-findings-vsts/issues"
   },
-  "homepage": "https://github.com/parasoft/parasoft-findings-vsts#readme"
+  "homepage": "https://github.com/parasoft/parasoft-findings-vsts#readme",
+  "devDependencies": {
+    "@types/q": "^1.5.5",
+    "@types/node-fetch": "^2.6.5",
+    "license-checker-rseidelsohn": "^4.2.5"
```
Comment (yshi-parasoft) [inline CodeCoverageQualityGate/package.json:28]: The library for generating licenses has been added, but the corresponding npm run script has not. I believe it is also a good practice to include the license in the same pull request whenever a new dependency is added. Therefore, please add a script for running the npm command to generate licenses and create a licenses folder.

**Pull request merged**